### ISSUE TITLE
`typeOf` returns a type that IMO is too long

### ISSUE THREAD

**Issue by coffeemug** [rethinkdb/rethinkdb#1260]
```
r.now().typeOf() => "PSEUDOTYPE(TIME)"
```

I think the word "pseudotype" is leaking an unnecessary amount of information for the following reasons:
1. Users won't necessary understand (or care for) the difference between types and pseudotypes
2. If they do understand/care, I think it would be better to just return the type and document with types are pseudotypes elsewhere
3. Type comparison becomes a bit of a pain. I wanted to get all rows where a certain attribute is a date, and typing this whole string was very unwieldy.

TL;DR: I think we should return "TIME" instead.

Thoughts?
